### PR TITLE
[SPARK-49928][PYTHON][TESTS] Refactor plot-related unit tests

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Refactor plot-related unit tests.

### Why are the changes needed?
Different plots have different key attributes of the resulting figure to test against. The refactor makes the comparison easier.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Test changes.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by xinrong-meng** [comment]
Irrelevant tests failed, retriggering:
```
ERROR [3.661s]: test_listener_events (pyspark.sql.tests.streaming.test_streaming_listener.StreamingListenerTests.test_listener_events)

ERROR [0.145s]: test_streaming_progress (pyspark.sql.tests.streaming.test_streaming_listener.StreamingListenerTests.test_streaming_progress)
```

File: python/pyspark/sql/tests/plot/test_frame_plot_plotly.py [hunk 1]
```diff
@@ -48,79 +48,174 @@ def sdf3(self):
...
-
-        self.assertEqual(fig_data["xaxis"], "x")
-        self.assertEqual(list(fig_data["x"]), expected_x)
-        self.assertEqual(fig_data["yaxis"], "y")
-        self.assertEqual(list(fig_data["y"]), expected_y)
-        self.assertEqual(fig_data["name"], expected_name)
+    def _check_fig_data(self, fig_data, **kwargs):
+        for key, expected_value in kwargs.items():
+            if key in ["x", "y", "labels", "values"]:
+                converted_values = [v.item() if hasattr(v, "item") else v for v in fig_data[key]]
```
Comment (xinrong-meng) [inline python/pyspark/sql/tests/plot/test_frame_plot_plotly.py:54]: To accommodate the change in the representation of scalars in NumPy 2, see [NumPy 2.0 release notes](https://numpy.org/devdocs/release/2.0.0-notes.html#representation-of-numpy-scalars-changed).

**Comment by xinrong-meng** [comment]
cc @zhengruifeng @HyukjinKwon would you please review thanks!

**Comment by xinrong-meng** [comment]
We may later port those expected_fig_data dictionaries to a separate JSON file for easier auditing if the number of tests increases

File: python/pyspark/sql/tests/plot/test_frame_plot_plotly.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/tests/plot/test_frame_plot_plotly.py:54]: nit, shall we use `isinstance(v, np.generic)` for this purpose?

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/tests/plot/test_frame_plot_plotly.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/tests/plot/test_frame_plot_plotly.py:54]: we can use it with `has_numpy`

**Review by zhengruifeng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
Merged to master, thank you!

**Pull request closed without merging**